vasyl_varyshchuk/Week_2/task_1/func_in_threads.py

```diff
@@ -0,0 +1,98 @@
+""""
+This module returns random size of file name from list by processing elements in separate threads
+"""
+
+import random
+import Queue
+from threading import Thread
+
+
+number_of_threads = 5
+len_of_list = number_of_threads * 100
+
+
+def recognize_file_type(file_name):
+    """This function returns extension of file passing file name as parameter"""
+    _, extension = file_name.split('.')
+    return extension
+
+
+def generate_file_names_list():
+    """This function generate list of files with random names. Length of list is specified by global variable
+    len_of_list which value depend on number on thread."""
+    names = ['na', 'ko', 'mi', 'fo', 'ad', 're', 'du', 'ly', 'ke', 'su', 'bo', 'na']
+    extensions = ['.txt', '.zip', '.ini', '.jpg', '.ttf']
+    file_names = []
+
+    for i in range(len_of_list):
+        file_name = ''
+        for j in range(3):
+            index = names.index(random.choice(names))
+            file_name += names[index]
+        ext = extensions.index(random.choice(extensions))
```

Comment:
It's super complicated and useless block of code.

random.choice(extensions) returns you exactly one of the extensions. You don't need to get an index of this extension and extract it again using this index.

Reply:
According to your remarks, I changed this part of the code.